pyEpiabm/pyEpiabm/utility/exception_logger.py

```diff
@@ -0,0 +1,24 @@
+#
+# Decorator to log exceptions in function
+#
+
+import logging
+
+
+def log_exceptions(message: str = ""):
+    """Provides a try/except clause around the function it is applied to.
+    Logs the type of error, and name of the function, if an error occurs,
+    with an optional message appended on the end.
+
+    :param message: Optional message to append to default log text
+    :type message: str
+    """
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except Exception as e:
+                logging.exception(f"{type(e).__name__} in "
+                                  + f"{func.__qualname__}()" + message)
```

Comment:
```suggestion
                                  + f"{func.__qualname__}(): " + message)
```
So that you dont have to worry about this in your message

Reply:
Ok lets not do that